spatialos-sdk/examples/project-example/setup.rs

```diff
@@ -0,0 +1,81 @@
+use fs_extra::dir::{self, CopyOptions};
+use std::ffi::OsString;
+use std::fs;
+use std::path::PathBuf;
+use std::process::Command;
+use tap::*;
+
+pub fn do_setup(spatial_lib_dir: Option<PathBuf>, out_dir: PathBuf) {
+    let spatial_lib_dir = spatial_lib_dir
+        .or_else(|| std::env::var("SPATIAL_LIB_DIR").map(Into::into).ok())
+        .expect("--spatial_lib_dir argument must be specified or the SPATIAL_LIB_DIR environment variable must be set");
+
+    // Determine the paths the the schema compiler and protoc relative the the lib
+    // dir path.
+    let schema_compiler_path = spatial_lib_dir.join("schema-compiler/schema_compiler");
+    let protoc_path = spatial_lib_dir.join("schema-compiler/protoc");
+
+    // Calculate the various output directories relative to `out_dir`.
+    let bin_path = out_dir.join("spatialos/schema/bin");
+    let tmp_path = out_dir.join("tmp");
+
+    // Create the output directories if they don't already exist.
+    fs::create_dir_all(&bin_path).expect("Failed to crate spatialos/schema/bin");
```

Comment:
Would we want to ensure that these directories are empty before we start working on them?

Reply:
Which directories are you referring to? The output dir and the temp dir? I imagine it's fine if the output dir isn't empty (e.g. updating the generated files when schemas change), might be good to issue a warning if the temp dir already exists? Might be a better way to handle temp dirs in that case, though 🤔

Reply:
I'm unsure of the behaviour of the schema compiler if the output file already exists, I think it should just overwrite it, but can't guarantee of the top of my head.

Broadly it should be fine though :smile:

Reply:
When I run it on my machine it appears to overwrite the existing file (the timestamp is updated, at least) so 🤷‍♀️

Reply:
Seems legit 💥